pkg/rid/store/cockroach/store.go

```diff
@@ -46,6 +51,38 @@ type Store struct {
 	clock  clockwork.Clock
 }
 
+// NewStore returns a Store instance connected to a cockroach instance via db.
+func NewStore(ctx context.Context, db *cockroach.DB, logger *zap.Logger) (*Store, error) {
+	store := &Store{
+		db:     db,
+		logger: logger,
+		clock:  DefaultClock,
+	}
+
+	if err := store.CheckCurrentMajorSchemaVersion(ctx); err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to construct store instance for remote ID")
```

Comment:
We will ultimately fail to construct a store instance, but the specific operation that failed here was checking the schema version; I suggest an error message of "Remote ID schema version check failed", "Error during remote ID schema version check" or something like that.